test/test_deferral.rb

```diff
@@ -146,4 +146,17 @@ def close
     assert r1.closed?
     assert r2.closed?
   end
+
+  test 'pass arguments to defer' do
+    r1 = Resource.new
+    r2 = Resource.new
+    1.times do
+      r = r1
+      Deferral.defer(r){|r| r.close }
```

Comment:
Use non-`r` variable name as block parameter to show it's different from loop local variable.